src/main/java/net/fabricmc/tinyremapper/ClassInstance.java

```diff
@@ -148,10 +163,37 @@ public int getMrjVersion() {
 		return mrjVersion;
 	}
 
+	@Override
 	public String getSuperName() {
 		return superName;
 	}
 
+	@Override
+	public String getSignature() {
+		return signature;
+	}
+
+	@Override
+	public MemberHeader getField(String name, String desc) {
+		return this.resolve(MemberType.FIELD, MemberInstance.getFieldId(name, desc, desc == null));
+	}
+
+	@Override
+	public MemberHeader getMethod(String name, String desc) {
+		return this.resolve(MemberType.METHOD, MemberInstance.getMethodId(name, desc));
+	}
+
+	@Override
+	@SuppressWarnings("ALL")
+	public Iterable<MemberHeader> allMembers() {
+		return (Iterable) this.members.values();
```

Comment:
You could fix the nasty generics here doing `Iterable<? extends MemberHeader>` as the return type

Reply:
yea but then the api looks uglier, better to have the ugly in impl than in API

Reply:
It's an `Iterable` though, your normal use in a `for` loop wouldn't be any different. If it was returning `Collection` directly then I'd understand, as it would as it would stop you adding to it (not that you'd be able to in this particular instance anyway), but here it is just using generics as they were intended

Reply:
There are two things I want to comment,
1. I really don't think `allMembers` is a good name since it does not retrieve all members, it only retrieves the members defined in this class, not its interface or superclass.
2. Why use iterable, not collection. I personally think the collection is easier to work with than iterable.

Reply:
>  your normal use in a for loop wouldn't be any different.

I usually work with `Stream`. Can `Iterable` easily converted to stream? Sorry for the stupid question, I really never touched iterable before.

Reply:
I'd use `Collection<? extends TrMember> getMembers()`